akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -568,16 +572,26 @@ private[akka] class ShardRegion(
   def matchingRole(member: Member): Boolean =
     member.hasRole(targetDcRole) && role.forall(member.hasRole)
 
-  def coordinatorSelection: Option[ActorSelection] =
-    membersByAge.headOption.map(m => context.actorSelection(RootActorPath(m.address).toString + coordinatorPath))
-
   /**
    * When leaving the coordinator singleton is started rather quickly on next
-   * oldest node and therefore it is good to send the GracefulShutdownReq to
+   * oldest node and therefore it is good to send the Register and GracefulShutdownReq to
    * the likely locations of the coordinator.
    */
-  def gracefulShutdownCoordinatorSelections: List[ActorSelection] =
-    membersByAge.take(2).toList.map(m => context.actorSelection(RootActorPath(m.address).toString + coordinatorPath))
+  def coordinatorSelection: List[ActorSelection] = {
+    @tailrec def select(result: List[Member], remaining: immutable.SortedSet[Member]): List[Member] = {
+      if (remaining.isEmpty)
+        result
+      else {
+        val m = remaining.head
+        if (m.status == MemberStatus.Up)
+          m :: result
+        else
+          select(m :: result, remaining.tail)
+      }
+    }
+
+    select(Nil, membersByAge).map(m => context.actorSelection(RootActorPath(m.address).toString + coordinatorPath))
```

Comment:
Um, so always all nodes that are up with oldest first? Shouldn't it be capped for larger clusters?

Reply:
+1, in that case this could take a while.

Reply:
> always all nodes that are up with oldest first

no, that's not what this does. it collects nodes until it detects one that is Up and the result includes that single Up.
example
```
(Up, Up, Up) => (Up)
(leaving, Up, Up) => (Leaving, Up)
(leaving, Exiting, Up, Up) => (Leaving, Exiting, Up)
```

Reply:
Ah, I missed the early return and thought the `isEmpty` was the single termination. Gotcha.